### PR TITLE
UI component/register

### LINKED ISSUE #10: Create registration component
Create a React component containing a registration form

### LINKED ISSUE #15: Add frontend validation
Find and add a frontend validation library

### PR DESCRIPTION
**Pull request by nedvin**
Registration component created. I used **Material-UI** for layout components as we discussed. After some research I found a good validation library, **react-hook-form**, that makes it super easy to create forms with React as it handles both the input of data into the form as well as the validation. **yup** is used for validation schema.

[Material-UI](https://material-ui.com)
[React Hook Form](https://react-hook-form.com)
[yup](https://github.com/jquense/yup)

Closes #10 
Closes #15

### CODE REVIEW

File: frontend/src/Registration.jsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+			dateOfBirth: {
+				year: validatedData.birthYear,
+				month: validatedData.birthMonth,
+				day: validatedData.birthDay,
+			},
+			username: validatedData.username,
+			password: validatedData.password,
+		};
+
+		console.log(formattedRegistrationData);
```
Comment (joaxcar) [inline frontend/src/Registration.jsx:47]: I guess this is the "stub" action where the actual action will take place? we should probably decide on how to handle this in our "in progress code". Maybe a TODO comment. Or a function call that is not implemented.

File: frontend/src/Registration.jsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+			<form onSubmit={handleSubmit(onSubmit)}>
+				<Grid container spacing={2}>
+					<Grid item xs={12} sm={6}>
+						<TextField
+							label="First name"
+							name="firstName"
+							inputRef={register}
+							fullWidth
+							error={validationErrors.firstName ? true : false}
+							helperText={
```
Comment (joaxcar) [inline frontend/src/Registration.jsx:62]: This is where we should take a minute to decide on an "internationalization approach" before we add a lot of components to the project. How can these messages be translated?

**Review by joaxcar — APPROVED** [review]
I am approving this. Looks good! I have left two comments that we need to consider for the future.

Is the "package-lock.json" file missing from the front end ? I think it should be added

File: frontend/src/Registration.jsx (hunk 1 above)
Comment (nedvin) [reply frontend/src/Registration.jsx:47]: Good catch! That's reasonable.

I can see "package-lock.json" at least in this branch.

**Review by RW90 — APPROVED** [review]
Nice work! I agree with your choice of packages

**Pull request merged**